imperial_coldfront_plugin/views.py

```diff
@@ -36,11 +37,73 @@
     check_group_owner_or_superuser,
     user_already_has_hpc_access,
     user_eligible_for_hpc_access,
+    user_eligible_to_be_pi,
 )
 
 User = get_user_model()
 
 
+@login_required
+def research_group_terms_view(request: HttpRequest) -> HttpResponse:
+    """View for accepting T&Cs and creating a ResearchGroup.
+
+    Args:
+        request: The Http request including the user information.
+
+    Returns:
+        The relevant Http response, depending on the permissions and the type of
+        request.
+    """
+    graph_client = get_graph_api_client()
+    user_profile = graph_client.user_profile(request.user.username)
+    if not user_eligible_to_be_pi(user_profile) and not request.user.is_superuser:
+        return HttpResponseForbidden("Permission denied")
+
+    if request.method == "POST":
+        form = TermsAndConditionsForm(request.POST)  # use TermsAndConditionsForm
+        if form.is_valid():
+            # Autogenerate name
+            group_name = f"Research Group {request.user.username}"
+
+            # If the group already exist, we just return that one
```

Comment:
We also need a check that `request.user` is not a member of another group already.

Reply:
Why? Just checking if there's no use case for someone to be a member to two research groups?